src/api/transactions.coffee

```diff
@@ -220,12 +228,71 @@ exports.updateTransaction = (transactionId) ->
 
   transactionId = unescape transactionId
   updates = this.request.body
+  that = this
 
   try
-    yield transactions.Transaction.findByIdAndUpdate(transactionId, updates).exec()
-    this.body = "Transaction with ID: #{transactionId} successfully updated"
-    this.status = 'ok'
-    logger.info "User #{this.authenticated.email} updated transaction with id #{transactionId}"
+    yield transactions.Transaction.findByIdAndUpdate transactionId, updates, () ->
```

Comment:
Why did you need to change this to a callback instead of just having it as it was before. This should work the same way.